src/mesh/boundary_standard.cxx

```diff
@@ -481,6 +481,69 @@ void BoundaryDirichlet::apply(Field3D &f,BoutReal t) {
 	  }
 	}			
       }
+    } else if (loc == CELL_ZLOW) {
+      // Shifted in Z
+
+      for(; !bndry->isDone(); bndry->next1d()) {
+        // Calculate the X and Y normalised values half-way between the guard cell and grid cell
+        BoutReal xnorm = 0.5*(   mesh->GlobalX(bndry->x)  // In the guard cell
+                                 + mesh->GlobalX(bndry->x - bndry->bx) ); // the grid cell
+
+        BoutReal ynorm = 0.5*(   mesh->GlobalY(bndry->y)  // In the guard cell
+                                 + mesh->GlobalY(bndry->y - bndry->by) ); // the grid cell
```

Comment:
This was a bit surprising at first, but it's using the fact that we never have both `bx` and `by` non-zero, and so always does the correct thing. Is that right, @johnomotani ?

Reply:
@ZedThree that's right - and it wasn't me, all the boundary conditions use that pattern :wink: